tests/lib.rs

```diff
@@ -1,26 +1,21 @@
 use std::fs::File;
 use std::path::Path;
 use tiled::{
-    error::TiledError, layers::LayerData, map::Map, parse, parse_file, parse_tileset,
-    properties::PropertyValue,
+    error::TiledError, layers::LayerData, map::Map, properties::PropertyValue, tileset::Tileset,
 };
 
-fn read_from_file(p: &Path) -> Result<Map, TiledError> {
+fn parse_map_without_source(p: impl AsRef<Path>) -> Result<Map, TiledError> {
```

Comment:
What does `parse_map_without_source` really mean?

Reply:
Parses a map from the given path, without setting the `source` attribute in the Map struct.

Reply:
When is that useful?

Reply:
Nowhere, you're right, I just did a blind translation from the old function without thinking too much.

Reply:
Had to look at it from my computer; it is useful because tests check if maps with the same data but from different sources are equal. The `source` property of them needs to always be set to `None`, otherwise all equality checks will fail.

Reply:
Hmm, if that's really the only difference I would suggest to do that only in the tests.

But, in general I don't really understand the need for the `source` on `Map`. Wouldn't it be better, to resolve the paths to absolute paths during loading, so that after loading we no longer need to know from where a map was loaded?

Reply:
Hmm, good idea. We could move `source` to `Tileset` instead and use absolute paths there.